src/main/java/com/parasoft/demoapp/config/rabbitmq/RabbitMQConfig.java

```diff
@@ -33,23 +37,61 @@ public class RabbitMQConfig {
     @Value("${spring.rabbitmq.password}")
     private String password;
 
+
     @Bean
-    public Connection connection() throws Exception {
-        ConnectionFactory factory = new ConnectionFactory();
+    public ConnectionFactory connection() throws Exception {
+        CachingConnectionFactory factory = new CachingConnectionFactory();
         factory.setHost(rabbitMqHost);
         factory.setPort(rabbitMqPort);
         factory.setUsername(user);
         factory.setPassword(password);
-        return factory.newConnection();
+        return factory;
+    }
+
+    @Bean
+    SimpleRabbitListenerContainerFactory simpleRabbitListenerContainerFactory(org.springframework.amqp.rabbit.connection.ConnectionFactory connectionFactory) {
+        SimpleRabbitListenerContainerFactory simpleRabbitListenerContainerFactory = new SimpleRabbitListenerContainerFactory();
+        simpleRabbitListenerContainerFactory.setConnectionFactory(connectionFactory);
+        return simpleRabbitListenerContainerFactory;
+    }
+
+    @Bean
+    public MessageConverter messageConverter() {
+        return new Jackson2JsonMessageConverter();
+    }
+
+    @Bean
+    public Queue orderServiceSendToQueue() {
```

Comment:
Need to create queues in advance otherwise the PDA will not be startedup when delete all queues on RabbitMQ UI manually, it means when PDA is started up it must have corresponding queues in RabbitMq.